commands/bake.go

```diff
@@ -84,6 +84,10 @@ func runBake(dockerCli command.Cli, targets []string, in bakeOptions, cFlags com
 	if in.provenance != "" {
 		overrides = append(overrides, fmt.Sprintf("*.attest=%s", buildflags.CanonicalizeAttest("provenance", in.provenance)))
 	}
+	if v := os.Getenv("SOURCE_DATE_EPOCH"); v != "" {
+		// TODO: extract env var parsing to a method easily usable by library consumers
+		overrides = append(overrides, fmt.Sprintf("*.args.SOURCE_DATE_EPOCH=%s", v))
```

Comment:
While this does change the previous behavior - I think the override behavior makes more sense (if we'd be ok with making a breaking change). That way, we can have all user-provided options that aren't in the file, override the file (instead of having different behavior for environment variables or command line options).

If we don't want to break, I can find another way to capture and push the env vars down.

Reply:
I don't think `SOURCE_DATE_EPOCH` is explicitly "override what I have in the file". It is more like "auto-load from the current environment because I can't find any other value". If the user wants override behavior, they can use `--set` or define a variable block in the bake file.

Reply:
Ah right that makes sense for the override behavior with bake.